python/xgbserver/xgbserver/__main__.py

```diff
@@ -32,5 +34,13 @@
 
 if __name__ == "__main__":
     model = XGBoostModel(args.model_name, args.model_dir, args.nthread)
-    model.load()
-    kfserving.KFServer().start([model]) #pylint:disable=c-extension-no-member
+    try:
+        model.load_from_model_dir()
+    except Exception as e:
+        ex_type, ex_value, _ = sys.exc_info()
+        logging.error(f"fail to load model {args.model_name} from dir {args.model_dir}. "
+                      f"exception type {ex_type}, exception msg: {ex_value}")
+        model.ready = False
+    # if fail to load model, start kfserver with an empty model list
```

Comment:
I think here we might want to check if the model repository is empty and start with empty list